### PR TITLE
Tidy up GitCoreRepository

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -4,7 +4,7 @@ plugins {
 
   id 'com.github.ben-manes.versions' version '0.28.0'
 
-  id 'com.diffplug.gradle.spotless' version '4.5.0' apply false
+  id 'com.diffplug.gradle.spotless' version '4.3.0' apply false
```
Comment (mkondratek) [inline build.gradle:7]: Why is that?

File: gitCoreApi/src/main/java/com/virtuslab/gitcore/api/IGitCoreBranchSnapshot.java [hunk 2]
```diff
@@ -22,25 +25,25 @@
    */
   String getFullName();
 
-  IGitCoreCommit derivePointedCommit() throws GitCoreException;
+  IGitCoreCommit getPointedCommit();
```
Comment (mkondratek) [inline gitCoreApi/src/main/java/com/virtuslab/gitcore/api/IGitCoreBranchSnapshot.java:28]: aweseme!

File: build.gradle (hunk 1 above)
Comment (PawelLipski) [reply build.gradle:7]: Idk why exactly, but 4.5.0 sometimes seemed to ignore un-formatted files (both in `apply` and in `check`)

Comment (PawelLipski) [reply build.gradle:7]: Possibly the combination of Gradle version (6.4) and plugin version was responsible? dunno

**Pull request merged**